site/content/docs/04-preloading.md

```diff
@@ -132,3 +132,23 @@ You can abort rendering and redirect to a different location with `this.redirect
 	}
 </script>
 ```
+
+#### Typing the function
+
+If you use TypeScript and want to access the above context methods, TypeScript will thow an error and tell you that it does not know the type of `this`. To fix it, you need to specify the type of `this` (see the [official TypeScript documentation](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#specifying-the-type-of-this-for-functions)). We provide you with helper interfaces so you can type the function like this:
+
+```html
+<script context="module">
+	import type { Page, PreloadContext } from "@sapper/common";
+
+	export async function preload(this: PreloadContext, page: Page, session: any) {
```

Comment:
Do we also want to use the function typing in this example?

```suggestion
	export const preload: Preload = async function(this: PreloadContext, page: Page, session: any) {
```

Reply:
This would be duplicated typing. The idea is to do either one or the other. Using the function typing means you don't need typings for `this` and the params.

Reply:
ah, right. still I wonder which is the better way to demonstrate it. using the `Preload` type seems a bit safer (it means you didn't put the wrong type for any parameter and adds typing for the return value) and it's also more concise